conduit/pipeline/pipeline.go

```diff
@@ -257,8 +277,26 @@ func (p *pipelineImpl) Init() error {
 
 	// InitProvider
 	round := sdk.Round(p.pipelineMetadata.NextRound)
-	// Initial genesis object is nil--gets updated after importer.Init
-	var initProvider data.InitProvider = conduit.MakePipelineInitProvider(&round, nil)
+
+	// Initialize Telemetry
+	var telemetryClient telemetry.Client
+	if p.cfg.Telemetry.Enabled {
+		// If telemetry cannot be initialized, log a warning and continue
+		// pipeline initialization.
+		var telemetryErr error
+		telemetryClient, telemetryErr = p.initializeTelemetry()
+		if telemetryErr != nil {
+			p.logger.Warn("Telemetry initialization failed. Continuing without telemetry.")
+		}
+		// Try sending a startup event. If it fails, log a warning and continue
+		event := telemetryClient.MakeTelemetryStartupEvent()
+		if telemetryErr = telemetryClient.SendEvent(event); telemetryErr != nil {
```

Comment:
Currently, without config modifications, we'll always log a warning here.

Reply:
Why?

Reply:
The pipeline tries to initialize the client [based on the config file](https://github.com/algorand/conduit/pull/74/files/61f253077c38c6408854525e524bc88595ef6a2c#diff-71752f54b1d598e2902b12d3d654ca6fd80b11c9848da2390967c3664c173b69R204). Since the default config file doesn't specify a URI, we don't send an event for now. We can change the default config.yml when we have an OpenSearch production server ready or alternatively hardcode the URI/credentials in like we do in algod.

Reply:
Is it the `Telemetry initialization failed` warning, or `failed to send telemetry event`?

Let's default `Enabled` to false, so this block wouldn't be hit anyway.

Reply:
> Is it the Telemetry initialization failed warning, or failed to send telemetry event?

Currently you'll get both warnings - it might be too redundant so I'll change it to send one or the other.

> Let's default Enabled to false, so this block wouldn't be hit anyway.

Good point, will do 👍

Reply:
Try to come up with a design that lets us gracefully disable telemetry. Keep in mind that plugins are going to be grabbing the telemetryClient object and they wont be able to check for Enabled.

Reply:
> Try to come up with a design that lets us gracefully disable telemetry. Keep in mind that plugins are going to be grabbing the telemetryClient object and they wont be able to check for Enabled.

The telemetryClient object has its own `Enabled` bool: https://github.com/algorand/conduit/pull/74/files#diff-ada99fa4b32a319d368310f322561a3e2292fefcd48533e763dff3069fe64cf0R11

This reads the config file upon pipeline initialization, checks if the config file has set the telemetry enabled bool to true, and initializes the client object (which then sets the client's bool to true). I think the current implementation lets us dynamically disable telemetry on the client as well.